src/Tasks/GenerateResource.cs

```diff
@@ -3539,22 +3498,19 @@ public static bool TryCreateCodeDomProvider(TaskLoggingHelper logger, string str
             {
                 provider = CodeDomProvider.CreateProvider(stronglyTypedLanguage);
             }
+            catch (Exception e) when
 #if FEATURE_SYSTEM_CONFIGURATION
-            catch (ConfigurationException e)
-            {
-                logger.LogErrorWithCodeFromResources("GenerateResource.STRCodeDomProviderFailed", stronglyTypedLanguage, e.Message);
-                return false;
-            }
+             (e is ConfigurationException || e is SecurityException)
+#else
+             (e is SystemException)
```

Comment:
This will catch a lot more than just ConfigurationException, most exceptions derive from SystemException.  If anything I'd say just catch all exceptions and log an error and avoid the ifdef but that's cleanup.

Reply:
Please please please do not log just Exception.Message and some generic text for anything but a specific exception. This just makes bugs harder to diagnose

Reply:
Even when this is a catch around a single method call?  Nick: for background CodeDom is throwing a private ConfigurationErrorsException https://github.com/dotnet/corefx/blob/2fac89f63c644ab81d936d62c91e6003302e718d/src/System.CodeDom/src/System/CodeDom/Compiler/CodeDomProvider.cs#L215-L222.  I suspect @stephentoub did this to avoid the dependency on ConfigurationManger.  Catching SystemException is nearly as bad as catching Exception.  You could check the type name (hack) and rethrow, maybe file an issue for us to push ConfigurationErrorsException down and throw the public version.

Reply:
yes, my feedback was to both the diff as is and the suggestion of changing to Exception.

It's better if it's a single method call, but then we'll end up with a bug in that method that is obscured. More hair falls out every time a bloody program tells me "object reference is not set to an instance of an object" like I did something wrong.

Can we do `catch (Exception ex) when (ex.GetType().Name == "ConfigurationErrorsException")`

Reply:
Sure, and I'll file a corefx bug.

Reply:
dotnet/corefx#40456 filed.

Reply:
> I suspect @stephentoub did this to avoid the dependency on ConfigurationManger.

Yes, ConfigurationManager didn't exist in corefx at the time.